nncf/common/pruning/utils.py

```diff
@@ -360,6 +377,63 @@ def calculate_in_out_channels_in_uniformly_pruned_model(pruning_groups: List[Clu
     return tmp_in_channels, tmp_out_channels
 
 
+def calculate_in_out_channels_in_uniformly_pruned_model(pruning_groups: List[Cluster[PrunedLayerInfoBase]],
+                                                        pruning_rate: float,
+                                                        full_input_channels: Dict[str, int],
+                                                        full_output_channels: Dict[str, int],
+                                                        pruning_groups_next_nodes: Dict[int, List[str]]):
+    """
+    Imitates filters pruning by removing `pruning_rate` percent of output filters in each pruning group
+    and updating corresponding input channels number in `pruning_groups_next_nodes` nodes.
+
+    :param pruning_groups: A list of pruning groups.
+    :param pruning_rate: Target pruning rate.
+    :param full_input_channels:  A dictionary of input channels number in original model.
+    :param full_output_channels: A dictionary of output channels number in original model.
+    :param pruning_groups_next_nodes: A dictionary of next nodes of each pruning group.
+    :return Dictionary of new input channels number {node_name: channels_num}
+    :return Dictionary of new output channels number {node_name: channels_num}
+    """
+    def get_num_of_sparse_elements_by_node(node_name: str) -> int:
+        old_out_channels = full_output_channels[node_name]
+        return get_rounded_pruned_element_number(old_out_channels, pruning_rate)
+
+    return _calculate_in_out_channels(pruning_groups,
+                                      get_num_of_sparse_elements_by_node,
+                                      full_input_channels,
+                                      full_output_channels,
+                                      pruning_groups_next_nodes)
+
+
+def calculate_in_out_channels_by_masks(pruning_groups: List[Cluster[PrunedLayerInfoBase]],
+                                       masks: Dict[str, NNCFTensor],
+                                       tensor_processor: NNCFBaseTensorProcessor,
+                                       full_input_channels: Dict[str, int],
+                                       full_output_channels: Dict[str, int],
+                                       pruning_groups_next_nodes: Dict[int, List[str]]):
```

Comment:
Please add a type hint for the return value.

Reply:
Done